app/metadata/metadata_store.py

```diff
@@ -40,14 +41,18 @@ def __init__(self, user=None):
             self.user = current_user
 
     def store(self, key, value):
-        data = self.user.get_questionnaire_data()
-        data[key] = value
+        if self.user is not None:
```

Comment:
I wonder if we re-write this class and initialize all the variables on construction. That way it could be immutable and we wouldn't need these checks.